### ISSUE TITLE
feat(iroh-relay): add rate limiting & configuration for rate limiting to the QUIC endpoint in the relay server

### ISSUE THREAD

**Issue by ramfox** [n0-computer/iroh#2967]

**Comment by Arqu** [comment]
Is this close-able now with https://github.com/n0-computer/iroh/pull/2961 merged?

**Comment by flub** [comment]
I don't think so, I think this issue is about the number of connections and connection rate made to the relay server for each service.  #2961 is rate-limiting the incoming bandwidth of established connections only.
